Audio HAL Improvements

akodanka opened the pull request.

kkm123 reviewed (approved):
Looks good to me!

sysopenci reviewed (commented):
Autobuild started from pull-request-changes on this PR.

Cactus automatically verified this PR.

For more information, see: /absp/builders/celadon_pmr0-autobuild/builds/103

akodanka committed:
Fix error when fetching timestamp for PA scenario

HAL reports error on get_presentation_position API,
when getTimestamp is called at framework.
This is because, the get_presentation_position API fails,
when there is no valid PCM device, which is the case
while using PulseAudio.

Tracked-On: OAM-89590
Signed-off-by: akodanka <anoob.anto.kodankandath@intel.com>

akodanka committed:
Featurize PulseAudio usage in Audio HAL

Move all PulseAudio related code inside a Preprocessor flag.
Thus, the new feature of using PulseAudio socket
can be enabled/disabled as required.

Tracked-On: OAM-89591
Signed-off-by: akodanka <anoob.anto.kodankandath@intel.com>

sysopenci reviewed (commented):
Failure when executing preintegration builder

FAILURE:Trigger Builds Triggered: build-cel - r1 (cel_apl), build-cel - r1 (cel_kbl), build-cel - r1 (celadon), build-cel - r1 (clk), build-cel - r2 (r2_cel_apl), build-cel - r2 (r2_cel_kbl), build-cel - r2 (r2_celadon), build-cel - r2 (r2_clk) - extra

More details, logs, and binaries: /absp/builders/celadon_pmr0-preintegration/builds/161

sysopenci reviewed (commented):
PR integrated into 'preintegration' builder



More details, logs, and binaries: /absp/builders/celadon_pmr0-preintegration/builds/162

Done by "Kodankandath, Anoob Anto" <anoob.anto.kodankandath@intel.com>

sysopenci reviewed (commented):
PR integrated into 'premerge' builder



More details, logs, and binaries: /absp/builders/celadon_pmr0-premerge/builds/74

Done by "Kodankandath, Anoob Anto" <anoob.anto.kodankandath@intel.com>

sysopenci reviewed (commented):
cactus finished building and testing your changes 

It was requested for merge by "Kodankandath, Anoob Anto" <anoob.anto.kodankandath@intel.com> 

so we merge it. Thanks for your contribution.

more details, logs, and binaries: /absp/builders/celadon_pmr0-mergerequest/builds/76
